newIDE/app/src/UI/Carousel.js

```diff
@@ -0,0 +1,375 @@
+          <Skeleton variant="rect" height={imageHeight} width={cellWidth} />
+        );
+      }
+    },
+    [renderImage, cellWidth, imageHeight]
+  );
+
+  const renderItemTitle = React.useCallback(
+    (item: ThumbnailType | SkeletonThumbnail): ?React.Node => {
+      if (!displayItemTitles) return null;
+      return (
+        <>
+          <Spacer />
+          {item.title ? (
+            <div style={{ ...styles.itemTitleContainer, width: cellWidth }}>
+              <Text noMargin style={styles.itemTitle}>
+                {item.title}
+              </Text>
+            </div>
+          ) : (
+            <Skeleton
+              variant="rect"
+              height={titleHeight}
+              width={(cellWidth / 3) * (1 + 2 * Math.random())} // Make rectangles of different lengths so that the UI feels more "alive".
+            />
+          )}
+        </>
+      );
+    },
+    [cellWidth, titleHeight]
+  );
+
+  const computeScroll = React.useCallback(
+    (
+      direction: 'left' | 'right',
+      scrollViewElement: HTMLUListElement
+    ): number => {
+      const visibleThumbnailsCount = Math.floor(
+        scrollViewElement.offsetWidth / widthUnit
+      );
+      const scale = visibleThumbnailsCount * widthUnit;
+
+      const currentScroll = scrollViewElement.scrollLeft;
+      const currentFirstVisibleItemIndex = currentScroll / widthUnit;
+
+      if (
+        direction === 'right' &&
+        currentFirstVisibleItemIndex >
+          itemsToDisplay.length - visibleThumbnailsCount - 1
+      )
+        return 0;
+      return roundScroll(
+        scrollViewElement.scrollLeft + scale * (direction === 'left' ? -1 : 1)
+      );
+    },
+    [widthUnit, itemsToDisplay]
+  );
+
+  const roundScroll = React.useCallback(
+    (value: number): number => {
+      return Math.round(value / widthUnit) * widthUnit;
+    },
+    [widthUnit]
+  );
+
+  const onClickArrow = (direction: 'left' | 'right') =>
+    React.useCallback(
+      (): void => {
+        const scrollViewElement = scrollView.current;
+        if (!scrollViewElement) return;
+        const newScrollPosition = computeScroll(direction, scrollViewElement);
+
+        scrollViewElement.scrollTo({
+          left: newScrollPosition,
+          behavior: 'smooth',
+        });
+      },
+      [computeScroll]
+    );
+
+  const handleScroll = React.useCallback(() => {
+    const scrollViewElement = scrollView.current;
+    if (!scrollViewElement) return;
+
+    if (!shouldDisplayLeftArrow)
+      setShouldDisplayLeftArrow(scrollViewElement.scrollLeft !== 0);
+  });
+  const handleScrollEnd = React.useCallback(
+    () => {
+      const scrollViewElement = scrollView.current;
+      if (!scrollViewElement) return;
+
+      scrollViewElement.scrollTo({
+        left: roundScroll(scrollViewElement.scrollLeft),
+        behavior: 'smooth',
+      });
+    },
+    [roundScroll]
+  );
+
+  const onFocusItem = event => {
+    event.target.scrollIntoViewIfNeeded();
```

Comment:
This will probably crash on Firefox which seems not to support this: https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoViewIfNeeded

Consider using https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView with `behavior: "smooth" `?

Reply:
I changed with scrollIntoView as you recommended it but without the behavior, not sure that it's what you want when navigating with tab